Change all machinery/attackby to use_tool

emmanuelbassil opened the pull request:
🆑 emmanuelbassil
tweak: Microwaves no longer accept solutions that contain forbidden ingredients. Feedback now given on which ingredient is incompatible.
bugfix: Un/Re-Anchoring a machine now properly updates machine's power usage.
/🆑 

So I know I said I'm going to try to avoid test merges but this is sadly an exception that will require one.


Some other things I did:
- Removed a bunch of attackby code governing wrenching/unwrenching anchored objects that was exactly similar to obj/use_tool. Now that it's handled by obj/use_tool; all you need to do is set the OBJ_ANCHORABLE flag to get this behavior. Tested the instances I did this. Did NOT add new behavior to any items.
- Small notice when examining an object that can be anchored/unanchored that it is possible to do so.
- attackby for newcaster handled damaging it. Axed this completely and made it use existing standardized health; works like a charm.
- Removed alot of 'add fingerprints' and 'add components'. These are handled by post_use_item and the parent obj/machinery/use_tool respectively; no need for repetition.
- Fixed as much styling as I could; some of this code is really old.
- Fixed an oversight on post_anchor_change where equipment power use did not change.
- Effectively moved alot of wrenching code out to post_anchor_change(). Did this for items with existing wrench/unwrench behavior under attackby() and no construction steps otherwise. No new behavior, all backend.
- Fixed oversight where pipe dispensers could work while unanchored and did not consume idle power.

emmanuelbassil committed:
attackby /machinery/a to use_tool

emmanuelbassil committed:
attackby /machinery/b-c to use_tool

Fixes

emmanuelbassil committed:
attackby /machinery/d-f to use_tool

emmanuelbassil committed:
attackby /machinery/g-o

emmanuelbassil committed:
attackby /machinery/p

emmanuelbassil committed:
attackby /machinery q-z

emmanuelbassil commented:
All machinery's attackbys have been changed.

Three things pending before it's ready:

- [ ] Do another run on wrenching/anchoring now that alot of is has been standardized.
- [ ] Fix some !user.unequip that I set to return TRUE where it should be FALSE
- [ ] Run through calls of the attackby proc by any machinery and replaced by use_tool

emmanuelbassil commented:
This is ready to go!

emmanuelbassil commented:
Fixed merged conflicts.

SierraKomodo commented on code/game/machinery/_machines_base/machine_construction/_construction.dm:
```diff
@@ -81,7 +81,7 @@
 	if(!validate_state(machine))
 		crash_with("Machine [log_info_line(machine)] violated the state assumptions of the construction state [type]!")
-		machine.attackby(I, user)
+		machine.use_tool(I, user)
```
Noting this may end up creating an infinite loop in certain conditions. Granted this logic already existed with attackby but it's worth looking at as maybe one of the things that's been plaguing us that we couldn't track the cause of.

`machine/use_tool()` calls `machine/component_attackby()`
`machine/component_attackby()` may call `machine_construction/attackby()`
`machine_construction/attackby()` may call `machine/use_tool()`

SierraKomodo commented on code/game/machinery/_machines_base/machinery.dm:
```diff
@@ -252,8 +252,9 @@
 
 
 /obj/machinery/post_anchor_change()
-	..()
+	update_use_power(anchored)
 	power_change()
+	..()
 
```
This technically would be a user facing change that should be changelogged.

SierraKomodo commented on code/game/machinery/_machines_base/machinery_components.dm:
```diff
@@ -236,14 +236,27 @@ GLOBAL_LIST_INIT(machine_path_to_circuit_type, cache_circuits_by_build_path())
 
+/obj/machinery/can_anchor(obj/item/tool, mob/user, silent)
+	if (use_power == POWER_USE_ACTIVE)
+		to_chat(user, SPAN_WARNING("Turn \the [src] off first!"))
```
```suggestion
        if (!silent)
		    to_chat(user, SPAN_WARNING("Turn \the [src] off first!"))
```

SierraKomodo commented on code/game/machinery/_machines_base/machinery_components.dm:
```diff
@@ -236,14 +236,27 @@ GLOBAL_LIST_INIT(machine_path_to_circuit_type, cache_circuits_by_build_path())
+
+
 /obj/machinery/post_anchor_change()
+	update_use_power(anchored)
 	power_change()
-	return ..()
+	..()
+
+/obj/machinery/post_anchor_change()
+	update_use_power(anchored)
+	power_change()
+	..()
 
```
`post_anchor_change()` is here twice

SierraKomodo commented on code/game/machinery/atmoalter/scrubber.dm:
```diff
@@ -213,10 +203,11 @@
 	base_type = /obj/machinery/portable_atmospherics/powered/scrubber/huge/stationary
 	machine_name = "large stationary portable scrubber"
 	machine_desc = "This is simply a large portable scrubber that can't be moved once it's bolted into place, and is otherwise identical."
+	obj_flags = null
```
```suggestion
	obj_flags = EMPTY_BITFIELD
```

SierraKomodo reviewed (dismissed):
Nowhere near thorough, only really had the time to look at a few files.

SierraKomodo commented on code/game/machinery/doors/airlock.dm:
```diff
@@ -968,35 +969,32 @@ About the new airlock wires panel:
-			..()
-		return
+	if (!repairing && MACHINE_IS_BROKEN(src) && locked) //bolted and broken
+		if (cut_bolts(C,user))
+			return TRUE
 
```
`cut_bolts()` includes feedback and visible messages and a timer, so this should return true regardless of the result of that proc to avoid doing two actions.

SierraKomodo commented:
Test merge is now live

emmanuelbassil commented:
Fixed reported microwave bug.
Should've returned a FALSE instead of TRUE since use_after (which was afterattack in the past) handles transfering reagents. Added comment saying as much

emmanuelbassil commented:
Fixed reported forensics bug. The way it worked before was resolve_attackby() was overriden at the forensics kit level to return FALSE if player was not on help intent. Since it returned FALSE; it went to afterattack. Since I switched afterattack to use_after and didn't fix the override; that behavior was inverted for the past month where if on a non-help intent they didn't work. And since they were in use_after; they also didn't work on doors and buttons where clicking with the sample kit would open the door/click the  button.


Switched the forensics interaction to use_before; and restored old behavior of only working on non-help intent. When on non-help; the fiber collection kit and print collection kit override all behavior including storage. Also added some information in the codex for both items re: necessity of being on non-help.

emmanuelbassil committed:
Check-paths

Checkpaths

Auto stash before rebase of "refs/heads/AssortedQI"

Fixes

Auto stash before rebase of "refs/heads/Wrapping"

Auto stash before rebase of "refs/heads/Attackby"

Auto stash before rebase of "refs/heads/Attackby"

SierraKomodo commented:
There doesnt appear to be anything game breaking with the machine attackby pr, so i'm good with approving and merging so we can return to dev on the main server and start looking at zombo

emmanuelbassil commented:
Merge conflicts fixed.
